cmd/bedEnrichments/bedEnrichments.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+)
+
+func bedEnrichments(method string, inFile string, secondFile string, noGapFile string, outFile string, verbose int, trimToRefGenome bool) {
+	var err error
+
+	if method != "exact" && method != "normalApproximate" && method != "upperBound" && method != "lowerBound" {
+		log.Fatalf("Error: unknown method. Found: %s.", method)
+	}
+
+	if verbose > 0 {
+		log.Println("Reading bed files.")
+	}
+
+	elementsOne := bed.Read(inFile)
+	elementsTwo := bed.Read(secondFile)
+	noGapRegions := bed.Read(noGapFile)
+
+	if trimToRefGenome {
```

Comment:
I think you should consider this being a separate function(s).  You could have a function call that makes the tree structure for the noGap, and then have two function calls that trim the two bed files to the noGap tree